rules/models/changes/ruleset.go

```diff
@@ -0,0 +1,96 @@
+package changes
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+)
+
+// UpdateOp is an update operation.
+type UpdateOp int
+
+// A list of supported update operations.
+const (
+	UnknownOp UpdateOp = iota
+	AddOp
+	RemoveOp
+	ChangeOp
+)
+
+// ParseUpdateOp parses an input string into an update op.
+func ParseUpdateOp(str string) (UpdateOp, error) {
+	switch str {
+	case "add":
+		return AddOp, nil
+	case "remove":
+		return RemoveOp, nil
+	case "change":
+		return ChangeOp, nil
+	default:
+		return UnknownOp, fmt.Errorf("unknown update op %s", str)
+	}
+}
+
+func (op UpdateOp) String() string {
+	switch op {
+	case AddOp:
+		return "add"
+	case RemoveOp:
+		return "remove"
+	case ChangeOp:
+		return "change"
+	default:
+		return "unknown"
+	}
+}
+
+// MarshalJSON marshals update op as JSON.
+func (op UpdateOp) MarshalJSON() ([]byte, error) {
+	return json.Marshal(op.String())
+}
+
+// UnmarshalJSON unmarshals data into update op.
+func (op *UpdateOp) UnmarshalJSON(data []byte) error {
+	var str string
+	if err := json.Unmarshal(data, &str); err != nil {
+		return err
+	}
+	res, err := ParseUpdateOp(str)
+	if err != nil {
+		return err
+	}
+	*op = res
+	return nil
+}
+
+// RuleSetChangeSet is a ruleset diff.
+type RuleSetChangeSet struct {
```

Comment:
We should call this `RuleSetChanges` to be consistent with what we use in swagger.

Reply:
I named them this way because you can have a list of ruleset changes for multiple namespaces and pluralization got a little awkward. So this would be `RuleSetChange` even though it contains more than one change and the struct on line 80 would be `RuleSetChanges`. Not a huge deal, I can make the change.

Reply:
Yeah was suggesting calling this one `RuleSetChanges` and the one on line 80 `RuleSetsChanges`.